main/main.cpp

```diff
@@ -2561,9 +2565,15 @@ Error Main::setup2() {
 
 	OS::get_singleton()->benchmark_begin_measure("scene");
 
+	// Initialize ThemeDB early so that scene types can register their theme items.
+	// Default theme will be initialized later, after modules and ScriptServer are ready.
+	initialize_theme_db();
+
 	register_scene_types();
 	register_driver_types();
 
+	register_scene_singletons();
```

Comment:
I trust moving the theme stuff around, but I'm not confident about this. How does this interact with the module/GDExtension scene initialization level? This code is _very_ core and moving stuff around typically breaks stuff.

Reply:
It's a method that I've added, it only registers ThemeDB right now and I moved it specifically so the singleton is available for other consumers as early as possible. 🙃